### ISSUE TITLE
Refactor step1 into multiple steps

### ISSUE THREAD

**Issue by zenalapp** [kathiehuang/haplotype_calling_pipeline#19]
Right now it runs fastq, trimmomatic, and mapping.

- [x] Split these into individual steps so that if you just want to re-do mapping you don't have to run all of the other steps again.
- [ ] Also make it so that you can run all the isolates in parallel (instead of running it in a loop within the script). This will make everything finish _much_ faster. 

The script is currently in perl, but if you're going to re-write a lot of it, it might be worth writing the individual re-factored scripts in Python, bash, or maybe R (depending on how much needs to be done outside of simply running the command).
